tdrs-backend/tdpservice/search_indexes/parsers/tanf_parser.py

```diff
@@ -0,0 +1,100 @@
+"""Transforms a TANF datafile into an search_index model."""
+
+import logging
+from ..models import T1  # , T2, T3, T4, T5, T6, T7, ParserLog
+# from django.core.exceptions import ValidationError
+from .util import get_record_type
+from .schema_defs.tanf import t1_schema
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+def active_t1_parser(line, line_number):
+    """Parse line in datafile as active case data, T1 only."""
+    family_case_schema = t1_schema()
+    # create search_index model
+    t1 = T1()
+    content_is_valid = True
+    for field in family_case_schema.get_all_fields():
+        if field.name == 'blank':
+            break
+        content = line[field.start-1:field.end]  # descriptor pdfs were off by one, could also adjust start values
+        if len(content) != field.length:
```

Comment:
I don't recall the path that prompted me to write this. It may have pre-dated me using the Field class. This could probably be removed I think but would double-check with the tests.

Reply:
TODO: Remove assuming it doesn't break tests